CHANGELOG.md

```diff
@@ -149,6 +156,7 @@ Although, Ktlint 0.47.0 falls back on property `disabled_rules` whenever `ktlint
 * Handle trailing comma in enums `trailing-comma` ([#1542](https://github.com/pinterest/ktlint/pull/1542))
 * Allow EOL comment after annotation ([#1539](https://github.com/pinterest/ktlint/issues/1539))
 * Split rule `trailing-comma` into `trailing-comma-on-call-site` and `trailing-comma-on-declaration-site` ([#1555](https://github.com/pinterest/ktlint/pull/1555))
+* Support globs containing directories in the ".editorconfig" supplied via CLI "--editorconfig"  ([#1551](https://github.com/pinterest/ktlint/pull/1551))
```

Comment:
This is a pretty big PR. Are all of these changes strictly dependent on each other? can we break it into smaller commits?

Reply:
Maybe it could have been splitted into multiple commits but it is pretty intertwined. Although I like smaller commits, I stopped doing that for two reasons: 1) almost none of my PR's are reviewed and 2) upon merging the commits are squashed.
I do no think that this PR could have been split into multiple smaller PR's.